### PR TITLE
Fix child validation and missing strand data validation

### PR DESCRIPTION
**Pull request by cortadocodes**
### Minor fixes and improvements

- [x] Fix child validation by adding missing argument `source`
- [x] Fix missing strand data validation by making strand attributes public (this removes the disparity between the attributes being private but the related names being used not including a leading underscore)
- [x] Replace  `uri_env_name` field with `backend` field in children schema

### CODE REVIEW

**Comment by cortadocodes** [comment]
@thclark this is required for https://github.com/octue/octue-sdk-python/pull/79. Are you happy with it?

File: twined/schema/children_schema.json [hunk 1]
```diff
@@ -8,15 +8,14 @@
...
-                "description": "The universally unique ID of the running child twin",
+                "description": "The universally unique ID (UUID) of the running child twin",
                 "type": "string"
             },
-            "uri_env_name": {
-                "description": "Name of the environment variable containing the URI of the twin (which may contain credentials)",
-                "type": "string",
-                "pattern": "^[A-Z]+(?:_[A-Z]+)*$"
+            "gcp_project_name": {
+                "description": "Name of the Google Cloud Platform (GCP) project the child exists in.",
```
Comment (thclark) [inline twined/schema/children_schema.json:15]: Need to think and talk more about this on monday.

 And it shouldn’t be provider-specific.

Maybe we need a “oneOf[object]” here allowing us to define a range of service backends: https, wss, pubsub, redis, etc. Of which the only one implemented is pubsub

In our case it’ll be an object containing

Comment (cortadocodes) [reply twined/schema/children_schema.json:15]: Sounds good. I also haven't dealt with the GCP credentials yet - should they go in the `credentials` strand or be associated with each child separately? Different children could be from different organisations on GCP and so need different credentials, or they could potentially be on different providers

File: twined/schema/children_schema.json [hunk 2]
```diff
@@ -8,15 +8,34 @@
...
+                "type": "object",
+                "oneOf": [
+                    {
+                        "type": "object",
+                        "title": "GCP Pub/Sub",
+                        "properties": {
+                            "name": {
+                                "description": "Type of backend (in this case, it should be GCPPubSub",
+                                "type": "string"
+                            },
```
Comment (thclark) [inline twined/schema/children_schema.json:25]: You could set the pattern of this to be a regex to enforce "GCPPubSub"
https://json-schema.org/draft-07/json-schema-validation.html#rfc.section.6.3.3

**Review by thclark — APPROVED** [review]

**Pull request merged**